src/libraries/System.Reflection.Metadata/tests/Metadata/TypeNameTests.cs

```diff
@@ -87,6 +96,16 @@ public void ParserIsNotEnforcingRuntimeSpecificRules(string input)
             }
         }
 
+        [Theory]
+        [InlineData("Int32", "Int32")]
+        [InlineData("System.Int32", "System.Int32")]
+        [InlineData("System.Int32[]", "System.Int32[]")]
+        [InlineData("System.Int32\\[\\]", "System.Int32[]")]
+        [InlineData("System.Int32\\", "System.Int32\\")]
+        [InlineData("System.Int32\\\\[]", "System.Int32\\[]")]
```

Comment:
```suggestion
        [InlineData("System.Int32\\\\[]", "System.Int32\\[]")]
        [InlineData("System\\.Int32", "System.Int32")]
```
This validates that the API does not check that the escape sequence is valid.

Reply:
Added.